hafas_event.py

```diff
@@ -0,0 +1,138 @@
+import re
+from datetime import datetime, timedelta
+
+from helper import Helper
+from hosted import CONFIG
+from mapping import CATEGORY_MAPPING, COLOUR_MAPPING
+
+REMOVE = re.escape(CONFIG["remove_string"].strip())
+
+
+class HAFASEvent:
+    def __init__(self, data):
+        self.json = data
+        self.duplicate = False
+        self.follow = None
+
+        self.id = data["JourneyDetailRef"]["ref"]
+
+        for product in data["Product"]:
+            if product.get("name") and product.get("catCode"):
+                self.symbol = product["name"]
+                self.category = product["catCode"]
+                self.operator = product.get("operatorCode", None)
+                self.icon = product.get("icon", None)
+                break
+        else:
+            self.symbol = ""
+            self.category = -1
+            self.operator = None
+            self.icon = None
+
+        if CONFIG["api_provider"] in CATEGORY_MAPPING:
+            self.category_icon = CATEGORY_MAPPING[CONFIG["api_provider"]].get(
+                str(self.category), ""
+            )
+        else:
+            self.category_icon = ""
+
+        scheduled = datetime.strptime(
+            data["date"] + " " + data["time"], "%Y-%m-%d %H:%M:%S"
+        )
+        if "rtTime" in data and "rtDate" in data:
+            self.realtime = datetime.strptime(
+                data["rtDate"] + " " + data["rtTime"], "%Y-%m-%d %H:%M:%S"
+            )
+            diff = self.realtime - scheduled
+            self.delay = int(diff.total_seconds() / 60)
+        else:
+            self.realtime = scheduled
+            self.delay = 0
+
+    def __lt__(self, other):
+        assert isinstance(other, HAFASEvent)
+        return self.realtime < other.realtime
+
+    def _clean(self, key):
+        return re.sub(
+            "^(" + REMOVE + "[- ])",
+            "",
+            re.sub(
+                "(\(" + REMOVE + "\))$", "", self.json[key].strip(), flags=re.IGNORECASE
+            ),
+            flags=re.IGNORECASE,
+        ).strip()
+
+    @property
+    def destination(self):
+        if "direction" in self.json:
+            return self._clean("direction")
```

Comment:
Maybe `self._clean()` should just return `None` instead of duplicating the code in here?

Reply:
Fixed in 7fe19b2.